drivers/gpio/Kconfig.mcp23x17

```diff
@@ -1,21 +1,39 @@
-# MCP23S17 GPIO configuration options
+# MCP23X17 GPIO configuration options
 
 # Copyright (c) 2020 Geanix ApS
+# Copyright (c) 2021 Prevas A/S
 # SPDX-License-Identifier: Apache-2.0
 
-menuconfig GPIO_MCP23S17
-	bool "MCP23S17 SPI-based GPIO chip"
-	depends on SPI
+menuconfig GPIO_MCP23X17
+	bool "MCP23017/MCP23S17 GPIO expander chip"
+	depends on (SPI || I2C)
 	select HAS_DTS_GPIO
 	help
-	  Enable driver for MCP23S17 SPI-based GPIO chip.
+	  Enable driver for MCP23017/MCP23S17 GPIO expander chip for either I2C or SPI.
+
+if GPIO_MCP23X17
+
+choice
+	prompt "MCP23x17 device select"
+	default MCP23X17_DEV_TYPE_I2C
+	help
+	  Select interface to communicate with MCP23x17 GPIO expander.
+	  MCP23017 is using SPI interface
+	  MCP23S17 is using I2C interface.
+
+config MCP23X17_DEV_TYPE_SPI
```

Comment:
I don't think we should need this.  Look at the bme280 sensor for an example of a device that supports different buses within the driver based solely on devicetree information.

Reply:
Also the devices use the exact opposite buses